### PR TITLE
Connect Report Details Backend

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*

*Description of changes:*
Connect Report Details API to front-end to provide actual Report details for each report in Reports table. 
`Note`: Depends on  #51 

`TODO`: Update data model/schema for Report details to return more information that's listed in the Report details page

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/components/main/report_details/report_details.tsx [hunk 1]
```diff
@@ -35,6 +35,7 @@ import { ShareModal } from './share_modal/share_modal';
 
 interface ReportDetailsRouteProps {
   reportId: string;
+  httpClient: any;
```
Comment (joshuali925) [inline public/components/main/report_details/report_details.tsx:38]: Why is this `any` instead of the core start type?

Comment (davidcui-amzn) [reply public/components/main/report_details/report_details.tsx:38]: Good catch- will change/see if this props interface can be removed

File: test/propsMock.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+ const propsMock = {
+   match: {
+     params: {
+       reportId: jest.fn(),
+     }
+   }
+ };
+
+ export default propsMock;
```
Comment (mihirsoni) [inline test/propsMock.js:24]: np: empty line,

File: public/components/main/report_details/report_details.tsx [hunk 3]
```diff
@@ -75,36 +50,52 @@ export const ReportDetailsComponent = (props) => {
...
-  schedule_details: '--',
-  alert_details: '--',
-  channel: 'Kibana Reports',
-  kibana_recipients: 'admin',
-  email_recipients: '--',
-  email_subject: '--',
-  email_body: '--',
-  report_as_attachment: false,
-};
+  const getReportDetailsData = (data) => {
```
Comment (mihirsoni) [inline public/components/main/report_details/report_details.tsx:62]: convert it to camelcase.

File: public/components/main/report_details/report_details.tsx [hunk 4]
```diff
@@ -75,36 +50,52 @@ export const ReportDetailsComponent = (props) => {
...
-  channel: 'Kibana Reports',
-  kibana_recipients: 'admin',
-  email_recipients: '--',
-  email_subject: '--',
-  email_body: '--',
-  report_as_attachment: false,
-};
+  const handleReportDetails = (e) => {
+    setReportDetails(e);
+  };
```
Comment (zhongnansu) [inline public/components/main/report_details/report_details.tsx:59]: if a function only contains another function, can you merge into one?

Comment (davidcui-amzn) [reply public/components/main/report_details/report_details.tsx:59]: Wrapping useState setter in a handler is standard practice

**Pull request merged**